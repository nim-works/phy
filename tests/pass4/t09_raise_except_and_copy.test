discard """
  description: "Ensure that copying at `Raise` exits works"
"""
(TypeDefs
  (Int 8)
  (ProcTy (Type 0))
  (ProcTy (Void))
)
(GlobalDefs)
(ProcDefs
  (ProcDef (Type 1) (Continuations
    (Continuation (Params) (Locals)
      (Continue 1 (IntVal 100) (List))
    )
    (Continuation (Params (Type 0)))
  ))
  (ProcDef (Type 2) (Continuations
    (Continuation (Params)
      (Locals (Type 0))
      (SelectBool (IntVal 0)
        (Continue 1 (List (Move (Local 0))))
        (Continue 2 (List (Move (Local 0))))
      )
    )
    (Continuation (Params (Type 0)) (Locals (Type 0))
      (Raise (IntVal 100)
        (Continue 3 (List (Move (Local 1)))))
    )
    (Continuation (Params (Type 0)) (Locals)
      (Raise (IntVal 100)
        (Continue 3 (List (Move (Local 0)))))
    )
    (Except (Params (Type 0) (Type 0)) (Locals)
      (Continue 4 (List))
    )
    (Continuation (Params))
  ))
)