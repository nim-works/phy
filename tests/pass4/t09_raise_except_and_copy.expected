(TypeDefs
  (Int 8)
  (ProcTy (Type 0))
  (ProcTy (Void))
)
(GlobalDefs)
(ProcDefs
  (ProcDef (Type 1) (Locals)
    (Continuations
      (Continuation (Params) (Locals)
        (Continue 1 (IntVal 100))
      )
      (Continuation
        (Params (Type 0))
      )
    )
  )
  (ProcDef (Type 2)
    (Locals (Type 0) (Type 0) (Type 0))
    (Continuations
      (Continuation (Params)
        (Locals (Local 0))
        (SelectBool (IntVal 0)
          (Continue 1)
          (Continue 2)
        )
      )
      (Continuation (Params)
        (Locals (Local 0) (Local 1))
        (Raise (IntVal 100)
          (Continue 4)
        )
      )
      (Continuation (Params)
        (Locals (Local 0))
        (Raise (IntVal 100)
          (Continue 3)
        )
      )
      (Except (Local 2)
        (Locals (Local 0) (Local 2) (Local 1))
        (Asgn (Local 1)
          (Copy (Local 0))
        )
        (Raise
          (Copy (Local 2))
          (Continue 4)
        )
      )
      (Except (Local 0)
        (Locals (Local 0) (Local 1))
        (Continue 5)
      )
      (Continuation (Params))
    )
  )
)
